### ISSUE TITLE
Request: Arranger Tree

### ISSUE THREAD

**Issue by newboerg** [fesch/Structorizer.Desktop#305]
It would be nice if the Structorizer had a Tree-View of the current "Project".

If you have more then 2 bigger Diagrams in the Arranger you can barely review anything
![problemclipboard02](https://cloud.githubusercontent.com/assets/7376421/21090605/2407aa9a-c041-11e6-9f81-d24a29547fa9.jpg)

I am proposing something like this:
![proposal](https://cloud.githubusercontent.com/assets/7376421/21090140/e2795eb4-c03d-11e6-876c-903b5da343a6.jpg)

**Comment by codemanyak** [comment]
You are right, at least a diagram index view showing the names (and possibly paths ) of the diagrams parked in Arranger would really be helpful.
Interestingly, about ten years ago, the early Structorizer version 1.26 seems to have conveyed a project concept already. I noticed that quite recently on scrolling through the changelog. Apparently, the subsequent Structorizer reimplementations did not re-introduce it. Maybe because a mandatory project context complicates the firts steps for a beginner. But a missing project context limits the usability for the advanced...
The Arranger index list might be a first helpful low-impact approach to address the needs of both. I think it might work as follows:

1. It appears only after some diagram has been sent to the Arranger.
2. It may be hidden or shown on demand, similar to the report list of the Analyser, but docked at the left or (possibly better) right margin.
3. A single click on an index entry would scroll the Arranger to the respective diagram and put it at top level.
4. A double click would fetch it back to the worl area of the Structorizer, replacing the currently edited diagram (after having offered to save its recent changes).

A further enhancement might be to pop up the file path of the diagram while the curser is hovering over it in the Arranger.
Would do you think? Would that help on the short term while we think about a possible specification of a project environment?

**Comment by newboerg** [comment]
This sounds exactly like what i had in mind :-)
The offering of save is especially appealing.
 Maybe add an Auto-Save-On-Change Option somewhere.

**Comment by codemanyak** [comment]
> The offering of save is especially appealing.
Maybe add an Auto-Save-On-Change Option somewhere.

Oops, I thought the Structorizer does already offer to save the current changes whenever another diagram is pushed into it? (I mentioned this in the "specification" above only for completeness).
Can you describe a situation where unsaved changes are overridden without a callback?
Thanks for your proposals and comments.

**Comment by newboerg** [comment]
Well, it happens when leaving a diagramm which has not been saved prior to execution.

**Comment by codemanyak** [comment]
> It should "Auto-Save" in that case, at least make it opt-in. its very annoying, because in case of using turtle it looses focus.

This hasn't been implemented in version 3.25-10 yet, I will file it as an extra enhancement issue since it requires some wide-spread modifications.

**Comment by codemanyak** [comment]
The focus problems described in #312 have successfully been addressed and will be fixed with the next version (3.25.11).

**Comment by codemanyak** [comment]
A keybinding for the `<del>` key has been added, allowing now to remove a diagram selected in the Arranger index from the Arranger. This way, the Arranger can easily be cleared from Structorizer. (Due with version 3.25-12 or release 3.26.)
